src/lifecycle/props-init.js

```diff
@@ -113,11 +98,9 @@ function createNativePropertyDefinition (name, opts) {
       opts.set(this, changeData);
     }
 
-    // Re-render on property updates if the should-update check passes.
-    if (prop.render(this, changeData)) {
-      const deb = this[$rendererDebounced] || (this[$rendererDebounced] = debounce(this.constructor[$renderer]));
-      deb(this);
-    }
+    // Queue a re-render.
+    const deb = this[$rendererDebounced] || (this[$rendererDebounced] = debounce(this.constructor[$renderer]));
```

Comment:
Performance wasn't a concern here. Now that you mention it, this should be able to be added when the element is upgraded rather than doing it lazily here.